### PR TITLE
Add Binomial distribution

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**
split from #4678 .

### CODE REVIEW

File: chainer/distributions/binomial.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        return self.n.data * self.p * (1 - self.p)
+
+
+@distribution.register_kl(Binomial, Binomial)
+def _kl_binomial_binomial(dist1, dist2):
+    if (dist1.n.data < dist2.n.data).any():
+        raise NotImplementedError()
+    n1 = dist1.n.data.astype(dist1.p.dtype)
+    xp = cuda.get_array_module(dist1.p)
+    is_inf = dist1.n.data > dist2.n.data
```
Comment (toslunar) [inline chainer/distributions/binomial.py]: - Do you really want to implement this case?
- KL = 0 if dist1.p = dist2.p = 0.

File: chainer/distributions/binomial.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+import chainer
+from chainer.backends import cuda
+from chainer import distribution
+from chainer.functions.array import broadcast
+from chainer.functions.array import where
+from chainer.functions.math import exponential
+from chainer.functions.math import lgamma
+import numpy
```
Comment (toslunar) [inline chainer/distributions/binomial.py]: Could you fix order of the imports?

File: tests/chainer_tests/distributions_tests/test_binomial.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+from chainer import distributions
+from chainer import testing
+import numpy
```
Comment (toslunar) [inline tests/chainer_tests/distributions_tests/test_binomial.py]: Could you fix order of the imports?

**Comment by toslunar** [comment]
Could you confirm the test results? @YoshikawaMasashi

**Comment by YoshikawaMasashi** [comment]
OK

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

**Pull request closed without merging**